sound/soc/sof/sof-client.c

```diff
@@ -382,6 +382,11 @@ void sof_client_ipc_rx_dispatcher(struct snd_sof_dev *sdev, void *msg_buf)
 		struct sof_ipc4_msg *msg = msg_buf;
 
 		msg_type = SOF_IPC4_NOTIFICATION_TYPE_GET(msg->primary);
+	} else {
+		dev_dbg_ratelimited(sdev->dev,
+				    "%s: Not supported IPC version: %d\n",
+				    __func__, sdev->pdata->ipc_type);
```

Comment:
not getting why this is a dev_dbg with rate limitation and the other below a warning.

Both are equally bad, aren't they?

Reply:
@plbossart, the `sof_client_register_ipc_rx_handler()` will happen once when a client tries to register for RX. Arguably this should not happen, but let's to be save.

The `sof_client_ipc_rx_dispatcher()` would be called for _all_ received notification (by the core), even if there is no client registered for any interest. I would not print anything if it were up to me, but it might comes as an incentive to implement proper handling of notification dispatch.
Would be called, since it is still up to the IPC code to call it and for IPC4 for example we are not doing it as we do not yet have any client which would need it. We have more important things to take care, I will revisit as soon as I got to it.

Reply:
use dev_dbg_once() then?